### PR TITLE
Binder redux

### PR DESCRIPTION
**Pull request by sjmiles**

### CODE REVIEW

File: pkg/Library/CoreXenon/Framework/Binder.js [hunk 1]
```diff
@@ -3,158 +3,105 @@
...
-
-export const addConnections = (layerId, connections, inputBindings) => {
-  // update input/outputBindings with cross-node connections
-  entries(connections).forEach(([propId, bound]) => {
-    const {id, prop} = Id.parsePropId(propId);
-    if (typeof bound === 'string') {
-      bound = [bound];
+  // unpack the node id
+  const nodeId = Id.sliceId(atomId, 0, -1);
+  // add input bindings from the atom spec
```
Comment (sjmiles) [inline pkg/Library/CoreXenon/Framework/Binder.js:38]: this bit has to deal with _implicit properties_

Comment (mariakleiner) [reply pkg/Library/CoreXenon/Framework/Binder.js:38]: 👍

File: pkg/Library/CoreXenon/Framework/Binder.js [hunk 2]
```diff
@@ -3,158 +3,105 @@
...
-  // from Atom-scope to App-scope.
-  const map = outputBindings[name];
-  if (map) {
-    const scoped = scopeOutput(output, map);
-    if (keys(scoped).length) {
-      log('output', keys(output), 'bound to', keys(scoped), 'for', name);
-    }
-    return scoped;
+const addBinding = (bindings, key, value) => {
+  const bound = bindings[key] ??= [];
```
Comment (mariakleiner) [inline pkg/Library/CoreXenon/Framework/Binder.js:59]: i really like how input and output bindings now have the same structure 🙏

File: pkg/Library/CoreXenon/Framework0/Id.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+/**
+ * @license
+ * Copyright 2023 NeonFlan LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
```
Comment (mariakleiner) [inline pkg/Library/CoreXenon/Framework0/Id.js:6]: this is `Framework0` - is this intentional?

**Review by mariakleiner — APPROVED** [review]

File: pkg/Library/CoreXenon/Framework0/Id.js (hunk 3 above)
Comment (sjmiles) [reply pkg/Library/CoreXenon/Framework0/Id.js]: Nope, ty.

File: pkg/Library/CoreXenon/Framework/Binder.js (hunk 2 above)
Comment (sjmiles) [reply pkg/Library/CoreXenon/Framework/Binder.js:59]: Me too, but I wonder about the cost here:

https://github.com/NeonFlan/xenonjs/pull/93/files#diff-068878bcbd2157fb1c3438099b44b2f0258b28620c8878794c346a72c9a51008R82

**Pull request merged**